### PR TITLE
[4.1] RavenDB-11571

### PR DESCRIPTION
**Pull request by aviv86**

### CODE REVIEW

File: test/SlowTests/Issues/RavenDB-11571.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+                using (var session = store.OpenSession())
+                {
+                    var documentCounters = session.CountersFor("users/1");
+                    var val = documentCounters.Get("Downloads");
+                    Assert.Equal(long.MaxValue, val);
+
+                    documentCounters.Increment("Downloads");
+                    var e = Assert.Throws<RavenException>(() => session.SaveChanges());
+                    Assert.Contains("Arithmetic operation resulted in an overflow", e.Message);
```
Comment (ayende) [inline test/SlowTests/Issues/RavenDB-11571.cs]: We need to tell on what counter / document this happened.

**Review by ayende — CHANGES_REQUESTED** [review]

File: test/SlowTests/Issues/RavenDB-11571.cs (hunk 1 above)
Comment (ppekrol) [reply test/SlowTests/Issues/RavenDB-11571.cs]: @aviv86 Let's create a CounterOverflowException for that then. We can push there document id, counter name, old value and new value and OverflowException as inner.

Comment (ppekrol) [reply test/SlowTests/Issues/RavenDB-11571.cs]: E.g. `Could not increment counter '{name}' from document '{documentId}' with value '{value}' by '{increment}'. Arithmetic operation resulted in an overflow.`

Comment (aviv86) [reply test/SlowTests/Issues/RavenDB-11571.cs]: done

File: src/Raven.Server/Documents/CountersStorage.cs [hunk 2]
```diff
@@ -317,11 +318,23 @@ public string IncrementCounter(DocumentsOperationContext context, string documen
...
+                    {
+                        value = checked(prev + delta); //inc
+
+                    }
+                    catch (OverflowException e)
+                    {
+                        throw new CounterOverflowException(
+                            $"Could not increment counter '{name}' from document '{documentId}' " +
+                            $"with value '{prev}' by '{delta}'. " +
+                            "Arithmetic operation resulted in an overflow.", e);
```
Comment (ayende) [inline src/Raven.Server/Documents/CountersStorage.cs:336]: The last sentence should be removed, that is the message of the overflow exception.

Comment (ppekrol) [reply src/Raven.Server/Documents/CountersStorage.cs:336]: Fixed

**Pull request merged**